### DISCUSSION TITLE
RSS autoposting

### DISCUSSION THREAD

**Discussion by jmduke** [buttondown/discussions#5] [Announcements]
## TL;DR

Connect one (or many) RSS feeds, supply a template of body/subject, and Buttondown takes care of the rest.

## The problem

A significant cohort of Buttondown's users use Buttondown as what I've started to call a "headless newsletter solution". What I mean by this is that they have their own CMS or blog or web presence and don't care about Buttondown's archives; they want to just use Buttondown to send emails and manage subscriptions. This is a pretty compelling use case, and I want to ameliorate it!

One pain point these users face is that there's a lot of administrivia involved in the actual 'sending emails' bit that can and should be automated. Buttondown's interface is lightweight, but logging in every week to send an email is still more friction that ideally needs to exist.

## The solution

RSS → emails! This is a solution that already exists in the realm of Zapier, but it seems so core to Buttondown's raison d'etre that I think it makes sense to build it in-house, especially because so many peer apps (Mailchimp, Substack) already have some version of this functionality. Here are some screenshots:

![image](https://user-images.githubusercontent.com/461046/212135957-eff897e7-d860-4a5d-9e39-b34edc2a3832.png)

![image](https://user-images.githubusercontent.com/461046/212135920-fb924c77-11aa-4214-8873-3cba589baf13.png)

The idea is, from a data modeling perspective, an `automation` object with a few user-supplied fields:

- url (what RSS feed are we pulling from?)
- behavior (what are we doing when we encounter new items?)
- cadence (how often is this automation run?)

## Timeline

This work is **actively developed**. I have folks using it in beta; email support@buttondown.email if you're interested in joining the beta program!

## Open questions

- [ ] How much customization is the right level of customization with regards to scheduling? Right now, I have it set as "every time a new feed item is posted", "weekly", or "monthly". I suspect there's a long tail of more esoteric schedules, but maybe we get lucky.
- [ ] Similarly, how much audience scope creep will accumulate over time? Right now there's just support for public emails without tags; I'm amenable to adding more, but I think there are going to be some interesting interface challenges there.

## Out of scope

(Nothing that I know of, for now!)

**Comment by JayCuthrell** [comment]
Howdy @jmduke  

Love this feature!

Have I found a bug or orphaned path? 

It appears that the Settings link I was using to manage RSS to Email has appeared then disappeared from Power Tools and the "settings-scroll-container" DIV.

<img width="315" alt="Screenshot 2023-10-31 at 3 43 15 PM" src="https://github.com/buttondown-email/discussions/assets/95317/ec6e054d-994f-4e12-8e99-8afaf87c2775">

I can temporarily get back to the correct link by re-adding an RSS feed then bookmarking the link and the "RSS Automation" DIV reappears (until my next login then it disappears again). 

[https://buttondown.email/settings#rss%20automation](https://buttondown.email/settings#rss%20automation)

<img width="293" alt="Screenshot 2023-10-31 at 3 49 48 PM" src="https://github.com/buttondown-email/discussions/assets/95317/c1003a48-5d6c-46c1-b271-bf817431592f">

**Reply by jmduke** [reply]
@JayCuthrell Definitely a bug! Converting this to an issue..
